bcda/auth/backend.go

```diff
@@ -48,7 +50,15 @@ func NewHash(source string) (Hash, error) {
 		return Hash(""), err
 	}
 
+	start := time.Now()
 	h := pbkdf2.Key([]byte(source), salt, hashIter, hashKeyLen, sha512.New)
+	hashCreationTime := time.Since(start)
+
+	if hashCreationTime < hashMinTime {
```

Comment:
What if it's taking too long? I think we should have an upper bound here as well. Maybe 3 sec?